delfin/tests/e2e/testdriver/__init__.py

```diff
@@ -0,0 +1,268 @@
+    """FakeStorageDriver shows how to implement the StorageDriver,
+    it also plays a role as faker to fake data for being tested by clients.
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        access_info = kwargs
+        if access_info is None:
+            raise exception.InvalidInput('Input access_info is missing')
+
+        self.array_json = access_info.get("extra_attributes").get("path")
+
+        with open(self.array_json) as f:
+            data = json.load(f)
+
+        # Verify Host & Port
+        f_host = data.get("access_info").get("rest").get("host")
+        f_port = data.get("access_info").get("rest").get("port")
+        f_user = data.get("access_info").get("rest").get("username")
+        f_pass = data.get("access_info").get("rest").get("password")
+        a_host = access_info.get("rest").get("host")
+        a_port = access_info.get("rest").get("port")
+        a_user = access_info.get("rest").get("username")
+        a_pass = access_info.get("rest").get("password")
+        a_pass = cryptor.decode(a_pass)
+        if f_host != a_host:
+            raise exception.InvalidIpOrPort
+        if f_port != a_port:
+            raise exception.InvalidIpOrPort
+        if f_user != a_user:
+            raise exception.InvalidUsernameOrPassword
+        if f_pass != a_pass:
+            raise exception.InvalidUsernameOrPassword
+
+    def reset_connection(self, context, **kwargs):
+        pass
+
+    def get_storage(self, context):
+        with open(self.array_json) as f:
+            data = json.load(f)
+            return data.get('storage')
+
+    def _return_json(self, key):
+        with open(self.array_json) as f:
+            data = json.load(f)
+            values = data.get(key)
+            for value in values:
+                value['storage_id'] = self.storage_id
+            return values
+
+    def list_storage_pools(self, ctx):
+        return self._return_json('storage_pools')
+
+    def list_volumes(self, ctx):
+        return self._return_json('volumes')
+
+    def list_controllers(self, ctx):
+        return self._return_json('controllers')
+
+    def list_ports(self, ctx):
+        return self._return_json('ports')
+
+    def list_disks(self, ctx):
+        return self._return_json('disks')
+
+    def list_quotas(self, ctx):
+        return self._return_json('quotas')
+
+    def list_filesystems(self, ctx):
+        return self._return_json('filesystems')
+
+    def list_qtrees(self, ctx):
+        return self._return_json('qtrees')
+
+    def list_shares(self, ctx):
+        return self._return_json('shares')
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def remove_trap_config(self, context, trap_config):
+        pass
+
+    @staticmethod
+    def parse_alert(context, alert):
+        pass
+
+    def clear_alert(self, context, alert):
+        pass
+
+    def list_alerts(self, context, query_para=None):
+        alert_list = [{
```

Comment:
we can move alert_list to a file just like it is done for storage (testdriver/storage.json) for better flexibility during testing.

Reply:
Alert Testing not considered in this PR. When it is added, we can simulate alert_list too.

Reply:
Done